modules/vpc-sc/main.tf

```diff
@@ -0,0 +1,157 @@
+  parent    = "organizations/${var.org_id}"
+  title     = each.key
+}
+
+resource "google_access_context_manager_access_level" "default" {
+  for_each = var.access_levels
+  parent   = "accessPolicies/${local.access_policy_name}"
+  name     = "accessPolicies/${local.access_policy_name}/accessLevels/${each.key}"
+  title    = each.key
+
+  dynamic "basic" {
+    for_each = try(toset(each.value.conditions), [])
+
+    content {
+      combining_function = try(each.value.combining_function, null)
+      conditions         {
+        ip_subnetworks   = try(basic.value.ip_subnetworks,null)
+        members          = try(basic.value.members,null)
+        negate           = try(basic.value.negate,null)
+      }
+    }
+  }
+}
+
+resource "google_access_context_manager_service_perimeter" "standard" {
+  for_each       = local.standard_perimeters
+  parent         = "accessPolicies/${local.access_policy_name}"
+  description    = "Terraform managed."
+  name           = "accessPolicies/${local.access_policy_name}/servicePerimeters/${each.key}"
+  title          = each.key
+  perimeter_type = each.value.type
+
+  # Enforced mode configuration
+  dynamic "status" {
+    for_each = each.value.enforced_config != null ? [""] : []
+
+    content {
+        resources           = formatlist("projects/%s", try(lookup(var.perimeter_projects, each.key, {}).enforced, []))
+        restricted_services = each.value.enforced_config.restricted_services
+        access_levels       = formatlist("accessPolicies/${local.access_policy_name}/accessLevels/%s", try(lookup(local.perimeter_access_levels_enforced, each.key, []), []))
+
+        dynamic "vpc_accessible_services" {
+            for_each = each.value.enforced_config.vpc_accessible_services != [] ? [""] : []
+
+            content {
+                enable_restriction = true
+                allowed_services = each.value.enforced_config.vpc_accessible_services
+            }
+        }
+    }
+  }
+
+  # Dry run mode configuration
+  use_explicit_dry_run_spec = each.value.dry_run_config != null ? true : false
+  dynamic "spec" {
+    for_each = each.value.dry_run_config != null ? [""] : []
+
+    content {
+      resources           = formatlist("projects/%s", try(lookup(var.perimeter_projects, each.key, {}).dry_run, []))
+      restricted_services = try(each.value.dry_run_config.restricted_services, null)
+      access_levels       = formatlist("accessPolicies/${local.access_policy_name}/accessLevels/%s", try(lookup(local.perimeter_access_levels_dry_run, each.key, []), []))
+
+
+      dynamic "vpc_accessible_services" {
+        for_each = try(each.value.dry_run_config.vpc_accessible_services != [] ? [""] : [],[])
+
+        content {
+          enable_restriction = true
+          allowed_services = try(each.value.dry_run_config.vpc_accessible_services, null)
+        }
+      }
+    }
+  }
+  
+  # Uncomment if used alongside `google_access_context_manager_service_perimeter_resource`, 
```

Comment:
after merge, let's discuss and see if it's worth using tw separate resource declarations for this resource, one with lifecycle and one without, to allow using the module from the repo without needing to edit

Reply:
Sure. I would not suggest to use google_access_context_manager_service_perimeter_resource. At the moment there is no option to specify if I want to add the project in `dry_run` or `enforced`.